### ISSUE TITLE
Problem with game analysis.

### ISSUE THREAD

**Issue by oservato** [WandererXII/lishogi#661]
Game analysis link: https://lishogi.org/study/wFff7t8y/uiZQ0y9F
80.P*45?? "Blunder".
Only LiShogi's Game Analysis says it's a serious mistake.
There is no reason for it to be a "blunder".
I did three game analyzes with LiShogi. He always says: "blunder".
I am "Gote" and the game analysis says: +99 (see graph).
Thank you.

**Comment by WandererXII** [comment]
Hey thanks for reporting it, I will look into it. It's definitely odd. Other parts of the analysis seem correct?
What do you mean by "I did three game analyzes with LiShogi. He always says: 'blunder'." ?

**Comment by oservato** [comment]
Thanks for the check.

On Lishogi ... I created three separate studies. I did a game analysis for each study. Lishogi always confirmed 80. P*45 blunder. I wanted to see if this was a temporary problem.

Other engines and players do not see the error. I add the graphs of ShogiGUI with different engines.
The other parts of LiShogi's game analysis are correct.
Thank you.
![1 LiShogi](https://user-images.githubusercontent.com/91876769/228483993-b6dd3c07-b9b8-48f2-a4fc-e2330fe30101.jpg)
![2 ShogiGUI - YO2018](https://user-images.githubusercontent.com/91876769/228484000-8b62ca16-2b2e-45f2-a457-d1cc124983b4.jpg)
![3 ShogiGUI - G2](https://user-images.githubusercontent.com/91876769/228484003-557da109-d709-4f55-8ea0-c8e7b766384f.jpg)
![4 ShogiGUI - S5](https://user-images.githubusercontent.com/91876769/228484005-0101aed2-518a-4c4c-ab8c-f4930a19ca9b.jpg)

**Comment by oservato** [comment]
https://lishogi.org/study/wFff7t8y/lawRe4em#0
I report another problem with the game analysis.
Move 98 and 100.
I attach graphics from LiShogi, ShogiGUI (Suisho 5 - YaneuraOu v7.5) and PlayShogi.
Thank you.
![lishogi 1](https://user-images.githubusercontent.com/91876769/230867413-5d0eba55-1ac9-4ab1-85ce-121992cd157e.jpg)
![playshogi 2](https://user-images.githubusercontent.com/91876769/230867417-b0a9a357-1366-4dfc-8700-1f1e60020a8e.jpg)
![shogigui 3](https://user-images.githubusercontent.com/91876769/230867419-0ae14799-2890-4986-8843-bda6395de320.jpg)

**Comment by Anfid** [comment]
Hello. I found a game that might be related to this issue. It is easily reproducible in both local analysis and server analysis: https://lishogi.org/Hqemz8e4

Moves [105](https://lishogi.org/Hqemz8e4#105) and [111](https://lishogi.org/Hqemz8e4#111) seem to represent a very similar behavior where engine line is only 1 move long and position evaluation jumps to -99.0. Also please note the positions before problematic moves, as it might be related to the issue. In all 3 games reported, we can see that problematic moves repeat the exact same position but after losing some material.

Analysis engine might have some shortcut to prevent repeated positions with losing material for analysis optimization on higher depths, so it just ends evaluation prematurely in these position? Hope this extra bit of info helps.

**Comment by WandererXII** [comment]
Thanks for more information. I found similar issue over at YaneuraOu - https://github.com/yaneurao/YaneuraOu/issues/139, but that should be solved. The fact that both server and local analysis has the same issue means it's either Yane or the eval file. The version and eval files are slightly different in server and local analysis, yet they seem to share the same issue. I will try some older versions and other eval files. One solution might also be to not send initial position and list of moves to YaneuraOu, but rather the current position without any moves, because then it works correctly... But then it wouldn't be able to detect repetitions (might be worth it?)

**Comment by WandererXII** [comment]
Just to update you guys. There are two issues hereL

1. The +/- 99, is because of YaneuraOu non-standard behavior, where when _superior position_ is reached it returns this high score. This is not a bug, but still something undesirable for us. 
2. The sudden 0 evaluation is because YaneuraOu marks position as repetition only after the second repetition, this is obviously to avoid evaluating the same positions over and over again. But still something we don't want.

From what I tested - shogiGUI doesn't feed YaneuraOu previous moves, just the positions, therefore YaneuraOu can't detect the repetitions and return the undesirable output. That's better than the current behavior and I could do this relatively quickly, but we still want some information about the repetitions, because there are cases when you want to repeat, or when the best move would be repeating and therefore you can't play it and YaneuraOu therefore shouldn't recommend it...

I opened a PR fixing these two issues - https://github.com/yaneurao/YaneuraOu/pull/266. If the PR is not accepted, I guess I will be using my fork of YaneuraOu for lishogi.

**Comment by WandererXII** [comment]
Forgot to close this, solved here:
https://github.com/yaneurao/YaneuraOu/commit/d49b90b23ff70ac93f7031d40170ff9eb50f8313
and here:
https://github.com/yaneurao/YaneuraOu/commit/aac87364db470dec505c1138d50508d322fb3fdd

If issue still happens please open a new issue.
